Automattic/wp-calypso
# Add support for `?coupon=[code]` in Calypso.

Depends on D14589-code.

This allows `?coupon=[code]` to be appended to any Calypso URL. The coupon code is saved in a cookie and applied at checkout automatically.

## Testing Instructions

- Apply diff: D14589-code
- Visit: https://calypso.live/?branch=add/coupon-urls
- Add something to your cart.
- From the checkout page where items are visible in your cart, instead of entering a coupon code, just add `?coupon=[code]` onto the end of the current URL and reload the page; e.g., `https://calypso.live/checkout/[mytestsite]?coupon=[code]`  
  (replace `[code]` with a valid coupon).
- Confirm the coupon is applied automatically and that your price is discounted.
  <img width="480" alt="2018-06-08_07-28-39" src="https://user-images.githubusercontent.com/1563559/41167008-34af8bac-6aee-11e8-9da0-04876700f225.png">
- Confirm that you have a `coupon` cookie now. You can type `document.cookie` into the browser console or install a cookie extension, such as [Cookie Inspector](https://chrome.google.com/webstore/detail/cookie-inspector/jgbbilmfbammlbbhmmgaagdkbkepnijn?hl=en).

---

- Clicking "Remove" should remove the coupon and delete the cookie automatically.
  <img width="503" alt="2018-06-08_07-34-45" src="https://user-images.githubusercontent.com/1563559/41167134-86e56220-6aee-11e8-8ff4-06f9fe5ec319.png">

---

- Make sure the `coupon` cookie has been deleted and empty your cart to start fresh.
- This time start from a different location with the coupon already in the URL from the very beginning; e.g., visit: https://calypso.live/?branch=add/coupon-urls&coupon= `[code]`
- Confirm again that you have the `coupon` cookie.
- Add something to your cart and proceed to checkout.
- Confirm the coupon is applied automatically.
  <img width="480" alt="2018-06-08_07-28-39" src="https://user-images.githubusercontent.com/1563559/41167008-34af8bac-6aee-11e8-9da0-04876700f225.png">


---

- Confirm no errors in browser console.
- Experiment further by interacting with the coupon field manually and fiddle with things a bit; i.e., try to break it or create an unexpected scenario. Please report any problems.
